### DISCUSSION TITLE
Format in Stackdriver format

### DISCUSSION THREAD

**Discussion by davidgargti20** [btraceio/btrace#544] [General]
I need to output log entries which can be used by stackdriver, which means json output, how can I do this in bTrace script.
Can anyone suggest a way to do that?

And also can we use logger in bTrace script?

**Comment by waihangcheung** [comment]
maybe make use of string api in BTraceUtils to compose & print out json

**Comment by jbachorik** [comment]
This is not readily available as of now. I have been playing with the idea of BTrace plugins or extensions that could be used to provide such extended functionality but so far there hasn't seemed to be much interest in that.

Anyway, for the time being you can request 'trusted' mode for your script like _@BTrace(trusted=true)_ and then you can use any class available on the classpath visible to the traced class. 

You could even extend the classpath with eg. a JSON serializer via 'bootClassPath' BTrace agent argument but then you must be extra careful not to introduce duplicate classes as that could affect the traced application.

**Reply by davidgargti20** [reply]
Thanks

**Reply by kushal-ti** [reply]
Hi, regarding this, if i were to use a external class visible to the traced application, say something like log4j, in my btrace script, trying to compile the btrace script gives an error. 
How do i get around this ?

**Reply by jbachorik** [reply]
You can do something like `./btracec -cp log4j.jar:other1.jar:other2.jar ...` - that should add the extra classes to the compiler classpath and you should be able to compile a BTrace probe referring to those classes.
